### PR TITLE
Update ref assembly source and fix some warnings

### PR DESCRIPTION
**Pull request by jeffkl**
* `ShouldCloneXmlAttribute()` method
* BuildRequestEngine.cs(843,17): warning CS1587: XML comment is not placed on a valid language element
* FileUtilities.cs(575): warning CS1574: XML comment has cref attribute 'GetFullPathNoThrow(string, string)' that could not be resolved
* FileUtilities.cs(597): warning CS1574: XML comment has cref attribute 'GetFullPathNoThrow(string, string)' that could not be resolved

### CODE REVIEW

File: src/XMakeBuildEngine/BackEnd/Components/BuildRequestEngine/BuildRequestEngine.cs [hunk 1]
```diff
@@ -840,14 +840,14 @@ private void CheckMemoryUsage()
...
-                /// results on very large builds, but build results are much more likely to be the bulk of
-                /// memory usage there.
+                // Since this causes synchronous I/O and a stop-the-world GC, it can be very expensive. If
+                // something other than build results is taking up the bulk of the memory space, it may not
+                // free any space. That's caused customer reports of VS hangs resulting from build requests
+                // that are very slow because something in VS is taking all of the memory, but every
+                // project build is slowed down by this codepath. To mitigate this, don't perform these
+                // checks in devenv.exe. On the command line, 32-bit MSBuild may still need to cache build
+                // results on very large builds, but build results are much more likely to be the bulk of
+                // memory usage there.
```
Comment (rainersigwald) [inline src/XMakeBuildEngine/BackEnd/Components/BuildRequestEngine/BuildRequestEngine.cs]: Oops 😬

**Comment by rainersigwald** [comment]
@dotnet-bot test Ubuntu14.04 Build for CoreCLR please

(Restore failed due to failure to get a lock?)

File: src/XMakeBuildEngine/BackEnd/Components/BuildRequestEngine/BuildRequestEngine.cs (hunk 1 above)
Comment (jeffkl) [inline src/XMakeBuildEngine/BackEnd/Components/BuildRequestEngine/BuildRequestEngine.cs]: I'll be turning on "Treat warnings as errors" soon 👍

**Comment by jeffkl** [comment]
Ruh roh

**Comment by rainersigwald** [comment]
@dotnet-bot test Ubuntu14.04 Build for CoreCLR please

**Pull request merged**